pkg/controller/storagecluster/reconcile.go

```diff
@@ -717,16 +779,37 @@ func (r *ReconcileStorageCluster) ensureCephCluster(sc *ocsv1.StorageCluster, re
 		statusutil.MapCephClusterNoConditions(&r.conditions, reason, message)
 	} else {
 		// Interpret CephCluster status and set any negative conditions
-		statusutil.MapCephClusterNegativeConditions(&r.conditions, found)
+		if sc.Spec.ExternalStorage.Enable {
+			statusutil.MapExternalCephClusterNegativeConditions(&r.conditions, found)
+		} else {
+			statusutil.MapCephClusterNegativeConditions(&r.conditions, found)
+		}
 	}
 
 	// When phase is expanding, wait for CephCluster state to be updating
 	// this means expansion is in progress and overall system is progressing
 	// else expansion is not yet triggered
 	if sc.Status.Phase == statusutil.PhaseClusterExpanding &&
+		!sc.Spec.ExternalStorage.Enable &&
 		found.Status.State != cephv1.ClusterStateUpdating {
 		r.phase = statusutil.PhaseClusterExpanding
 	}
+
+	if sc.Spec.ExternalStorage.Enable {
+		if found.Status.State == cephv1.ClusterStateConnecting {
+			sc.Status.Phase = statusutil.PhaseConnecting
+		} else if found.Status.State == cephv1.ClusterStateConnected {
+			sc.Status.Phase = statusutil.PhaseConnected
+		} else {
+			sc.Status.Phase = statusutil.PhaseNotReady
+		}
+
+		if err = r.client.Status().Update(context.TODO(), sc); err != nil {
+			reqLogger.Error(err, "failed to update ignored resource")
```

Comment:
Error message is incorrect. There is no Ignored resource here.

Reply:
```suggestion
			reqLogger.Error(err, "failed to update external cluster status")
```

Reply:
Yup, made the changes accordingly...